FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +43,53 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] httpClientResult in
+
+			guard self != nil else { return }
+
+			var result: FeedLoader.Result
+
+			let feedLoaderResult = Self.httpClientResult2RemoteFeedLoaderResult(httpClientResult)
+			switch feedLoaderResult {
+			case .success(let responseData):
+				result = Self.responseData2FeedLoaderResult(responseData)
+
+			case .failure(let error):
+				result = .failure(error)
+			}
+
+			completion(result)
+		}
+	}
+
+	private static func httpClientResult2RemoteFeedLoaderResult(_ httpClientResult: HTTPClient.Result) -> RemoteFeedLoaderResult {
+		let feedLoaderResult = httpClientResult.mapError { _ in Error.connectivity }
+			.flatMap { (responseData, httpResponse) -> RemoteFeedLoaderResult in
+				guard httpResponse.isStatusOK else {
+					return .failure(Error.invalidData)
+				}
+				return .success(responseData)
+			}
+		return feedLoaderResult
+	}
+
+	private static func responseData2FeedLoaderResult(_ responseData: Data) -> FeedLoader.Result {
+		var result: FeedLoader.Result
+		do {
+			let remoteFeedImages = try JSONDecoder().decode(ResponseRootEntity.self, from: responseData)
+			let feedImages = remoteFeedImages.items.map { $0.feedImage }
+			result = .success(feedImages)
+		} catch {
+			result = .failure(Error.invalidData)
+		}
+		return result
```

Comment:
There's no need to use a `do/catch` block if we're ignoring the error `thrown` by `JSONDecoder().decode`, we can replace it with an optional `try?`. Also, there's no need to create a variable `result`, we can return any `Result` immediately:
```suggestion
		if let remoteFeedImages = try? JSONDecoder().decode(ResponseRootEntity.self, from: responseData) {
			let feedImages = remoteFeedImages.items.map { $0.feedImage }
			return .success(feedImages)
		}
		return .failure(Error.invalidData)
```